bittide-shake/bin/Shake.hs

```diff
@@ -112,9 +99,16 @@ clashBuildDir = buildDir </> "clash"
 vivadoBuildDir :: FilePath
 vivadoBuildDir = buildDir </> "vivado"
 
-getConstraintFilePath :: TH.Name -> IO FilePath
-getConstraintFilePath target =
-  getDataFileName ("data" </> "constraints" </> nameBase target <> ".xdc")
+getConstraintFilePath :: String -> IO FilePath
+getConstraintFilePath target = do
+  out <- readProcess "cabal"
+    [ "run"
+    , "-v0"
+    , "terminal-api"
+    , "getDataFileName"
+    , "data/constraints" </> (snd $ splitName target) <> ".xdc"
```

Comment:
```suggestion
    , "data/constraints" </> snd (splitName target) <> ".xdc"
```

Function application takes precedence over _everything_, so it's fine to leave out the parentheses. (I'm usually on the side of "more parentheses more better", but given that function application is the absolute max you can get in Haskell it's fine.)